src/java.base/share/classes/java/util/zip/ZipCoder.java

```diff
@@ -38,7 +38,13 @@
 import sun.nio.cs.UTF_8;
 
 /**
- * Utility class for ZIP file entry name and comment decoding and encoding
+ * Utility class for ZIP file entry name and comment decoding and encoding.
+ * <p>
+ * The {@linkplain #UTF8 UTF-8 ZipCoder instance}, which gets returned by {@link #get(Charset)}
+ * for {@linkplain java.nio.charset.StandardCharsets#UTF_8 UTF-8 Charset} is thread-safe
+ * and can be used concurrently by multiple threads. All other {@code ZipCoder} instances
+ * are not thread-safe and external synchronization is required by callers, if those
+ * instances are to be used concurrently by multiple threads.
```

Comment:
I think you can trim this down to just say that the UTF-8 ZipCoder is thread safe, ZipCoder for other charsets require synchronization.

Reply:
Done - the PR has been updated to trim this text to what you suggested.